design/dedicated-osd-pod.md

```diff
@@ -0,0 +1,73 @@
+# Run OSD in its own Pod
+
+## TL;DR
+
+A one-OSD-per-Pod placement should be implemented to improve reliability and resource efficiency for Ceph OSD daemon.
+
+## Background
+
+Currently in Rook 0.7, Rook Operator starts a ReplicaSet for Rook OSD  on each storage node. The ReplicaSet has just one replica. Rook OSD scans and prepares devices, creates OSD IDs and data directories, generates Ceph configuration. At last, Rook OSD starts all Ceph OSD daemons in foreground and tracks the OSD processes.
+
+As observed, all Ceph OSDs are running in the same Pod.
+
+## Limitations
+
+One Pod for all OSDs doesn't have the highest reliability nor efficiency. If the Pod is deleted, accidentally or during maintenence, all OSDs are down till the ReplicaSet restart. If nodes have asymmetric configurations, some have more devices than others, the Rook Operator doesn't request appropriate resources for the OSD Pods, since the resources are requested before the OSDs are discovered.
+
+A more comprehensive discussions can be found at [this issue](https://github.com/rook/rook/issues/1341).
+
+## Proposal   
+
+We propose the following OSD startup change.
+
+### Create new OSDs
+| Sequence |Rook Operator  | Rook OSD   | Ceph OSD  | Note  |
+|---|---|---|---|---|
+| 1  | Read devices from cluster CRD and create a Pod for Rook OSD for each storage node  |   |   |  |
```

Comment:
I see the sequence as the following:
1. The Rook agent detects the devices and creates/updates the configmap per node
1. The user creates the cluster CRD. The operator starts configuring the cluster according to the crd, configures the mons, and then the OSDs.
1. For each device by the rook agents that matches the filter for the cluster, and is not already consumed by another cluster):
   1. Create a K8s batch job that will provision the device. The device will be formatted, partitioned, etc, and the osd will be initialized. 
   1. The status of the device in the configmap will be updated to `provisioned` with the osd ID.
   1. The operator creates a deployment that will run the osd. The name of the deployment/rs/pod will include the node name and osd ID for easier debugging later.
   1. The status of the device in the configmap will be updated to `started`